### PR TITLE
Created Models

### PR DESCRIPTION
**Pull request by wooglow**

### CODE REVIEW

**Review by HibiAdelei — APPROVED** [review]

File: models/spot.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            allowNull: false,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        title: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
+        latitude: {
+            type: DataTypes.DECIMAL(10.5),
```
Comment (LemonPocky) [inline models/spot.js]: This is invalid. Maybe you meant 10,5?

File: models/spot.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        longtitude: {
+            type: DataTypes.DECIMAL(10.5),
+            allowNull: false,
+            validate: {
+                isDecimal: true
+            }
+        },
+        description: {
+            type: DataTypes.STRING,
```
Comment (LemonPocky) [inline models/spot.js]: String only goes up to 255 characters, should be TEXT so the description can be longer.

File: models/spot.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                unique: "false"
+            }
+        }
+    },
+    {
+        sequelize,
+        timestamps: false,
+        freezeTableName: true,
+        underscored: true,
+        modelName: 'Spot',
```
Comment (LemonPocky) [inline models/spot.js]: Model names are case-sensitive, this needs to be lowercase.

File: models/spot.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            type: DataTypes.DECIMAL(10.5),
+            allowNull: false,
+            validate: {
+                isDecimal: true
+            }
+        },
+        description: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
```
Comment (LemonPocky) [inline models/spot.js:35]: spot requires a user_id foreign key because there is a 1-to-many relationship with user

File: models/tag.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+Tag.init(
+    {
+        id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        tag_name: {
+            type: DataTypes.STRING,
```
Comment (LemonPocky) [inline models/tag.js:16]: add not null to tag_name

File: models/user.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          10
+        );
+        return updatedUserData;
+      },
+    },
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: 'User',
```
Comment (LemonPocky) [inline models/user.js]: Model names are case-sensitive, this needs to be lowercase.

File: models/user.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      primaryKey: true,
+      autoIncrement: true,
+    },
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
```
Comment (LemonPocky) [inline models/user.js:24]: Users also require a TEXT field called 'bio' which stores a description about the user that is displayed on their profile.

File: models/index.js [hunk 8]
```diff
@@ -1,5 +1,31 @@
...
+});
+
+Spot.belongsToMany(Tag, {
+    through: SpotTag,
+    foreignKey: 'spot_id',
+});
+
+
+
 module.exports = { User };
```
Comment (LemonPocky) [inline models/index.js]: Make sure you export all the models here.

**Review by LemonPocky — CHANGES_REQUESTED** [review]

**Review by wooglow — COMMENTED** [review]
Change completed

**Review by LemonPocky — APPROVED** [review]

**Pull request merged**